transparentpath/gcsutils/transparentpath.py

```diff
@@ -1693,6 +1698,39 @@ def cast_iterable(self, iter_: Iterable):
             to_ret = [self.cast_iterable(item) for item in iter_]
             return to_ret
 
+    def caching_saver(self, file):
+        """
+        save fetched data in tmp file or dict, if not exceeding caching_max_memory else remove oldest data
```

Comment:
Indicate here how to set "caching" to use memoy or tmp. Also it is not "else", it will still save the new data if memory is exceeded. Also say that this is used when "read" is called, so it stores data and not paths.